### PR TITLE
fix: In App Purchases(IAP) Improvements

### PR DESCRIPTION
**Pull request by farhan-arshad-dev**
## Description

- Add value prop screen event.
- Remove value_prop_enabled check for IAP.
- Put restore button behind the IAP enable config.
- Make IAP error dialog Non-Cancellable.
- Hide the upgrade button for collapsed toolbar in course dashboard.
- Improve IAP full-screen dialog UI.

| Old  | Improved |
| ------------- | ------------- |
| Screenshot_20240724_184037  | Screenshot_20240724_194230  |

### CODE REVIEW

File: core/src/main/java/org/openedx/core/ui/IAPUI.kt [hunk 1]
```diff
@@ -183,7 +183,7 @@ fun NoSkuErrorDialog(
                 onClick = onConfirm
             )
         },
-        onDismissRequest = onConfirm
+        onDismissRequest = {}
```
Comment (HamzaIsrar12) [inline core/src/main/java/org/openedx/core/ui/IAPUI.kt:186]: FYI: We could have also set DialogProperties. 😉

File: core/src/main/java/org/openedx/core/presentation/dialog/IAPDialogFragment.kt [hunk 2]
```diff
@@ -228,6 +225,10 @@ class IAPDialogFragment : DialogFragment() {
         }
     }
 
+    override fun getTheme(): Int {
```
Comment (HamzaIsrar12) [inline core/src/main/java/org/openedx/core/presentation/dialog/IAPDialogFragment.kt:228]: My status bar color is this now.

File: core/src/main/java/org/openedx/core/presentation/dialog/IAPDialogFragment.kt
Comment (HamzaIsrar12) [inline core/src/main/java/org/openedx/core/presentation/dialog/IAPDialogFragment.kt:1]: The fullscreen loader dialog is still dismissable on the back press.

**Review by HamzaIsrar12 — CHANGES_REQUESTED** [review]
Minor nits

File: core/src/main/java/org/openedx/core/ui/IAPUI.kt (hunk 1 above)
Comment (farhan-arshad-dev) [reply core/src/main/java/org/openedx/core/ui/IAPUI.kt:186]: right, but `onDismissRequest` is mandatory for the definition of the AlertDialog that we are using.

Comment (farhan-arshad-dev) [reply core/src/main/java/org/openedx/core/ui/IAPUI.kt:186]: for the safe side I'm adding the properties.

**Comment by farhan-arshad-dev** [comment]
@HamzaIsrar12 PR is ready for another pass.

**Review by HamzaIsrar12 — APPROVED** [review]

**Pull request merged**